### PR TITLE
Generalized 4 external interface into 2 abstract interfaces

### PR DESCRIPTION
**Pull request by billwuqin**

### CODE REVIEW

File: draft-irtf-nmrg-network-digital-twin-arch.md [hunk 1]
```diff
@@ -846,7 +846,7 @@ It can include both physical entities and some virtual entities (e.g., vSwitches
 traffic and provide actual network services.
 All or subset of network elements in the real network deliver network data, directly or through other systems, to
 the NDT, through appropriate input interfaces. Network elements may receive control inputs, through specific output interfaces,
-from operations systems in which NDTs play a role.
+from operations systems in which NDTs play a role. The input interfaces and output interfaces might vary in functional NDT use cases.
```
Comment (boucadair) [inline draft-irtf-nmrg-network-digital-twin-arch.md:849]:
```suggestion
from operations systems in which NDTs play a role. The input and output interfaces might vary as a function of the specific NDT use case. Note that these interfaces are bidirectional interfaces.
```

**Comment by cfjanz** [comment]
This: “The input and output interfaces might vary as a function of the specific NDT use case”

Is tantamount to saying that what an NDT is, and does, is use case-dependent…. So that you need to define a different NDT independently for every use case. If you have different interfaces, you must also have different functions that stand behind those interfaces (e.g. if you have an NDT-as-controller as in current Figure 2, you must have intent translation, inner and outer closed loop, control and orchestration functions, etc. If you have an NDT for AI training, or personnel training, or visualization, or optimization … different interfaces AND different functions).

Why isn’t it easier and more sensible to have ONE definition of NDT? And then you build use cases by associating use case-dependent further functions with the NDT?

Chris

**Comment by cfjanz** [comment]
This said, it’s the Figure (Figure 2) more than detailed text like “the input and output interfaces may vary …” hey, they might, we’re not saying how, and we’re not defining interfaces in detail) that is the problem. Figure 3 and section 8 were added to show and explain more fulsomely the use case that Figure 2 was suggesting (“NDT as controller” -> NDT in/supporting controller). Personally I’d still rather just get rid of Figure 2. Given Figure 1, the current Section 7 text, and Figure 3/Section 8, I don’t think Figure 2 adds anything but confusion.

Chris


From: Christopher Janz
Sent: Tuesday, September 10, 2024 9:33 AM
To: 'cheneyzhoucheng/network-digital-twin' ***@***.***>; cheneyzhoucheng/network-digital-twin ***@***.***>
Cc: Review requested ***@***.***>
Subject: RE: [cheneyzhoucheng/network-digital-twin] Generalized 4 external interface into 2 abstract interfaces (PR #32)

This: “The input and output interfaces might vary as a function of the specific NDT use case”

Is tantamount to saying that what an NDT is, and does, is use case-dependent…. So that you need to define a different NDT independently for every use case. If you have different interfaces, you must also have different functions that stand behind those interfaces (e.g. if you have an NDT-as-controller as in current Figure 2, you must have intent translation, inner and outer closed loop, control and orchestration functions, etc. If you have an NDT for AI training, or personnel training, or visualization, or optimization … different interfaces AND different functions).

Why isn’t it easier and more sensible to have ONE definition of NDT? And then you build use cases by associating use case-dependent further functions with the NDT?

Chris

File: draft-irtf-nmrg-network-digital-twin-arch.md (hunk 1 above)
Comment (billwuqin) [reply draft-irtf-nmrg-network-digital-twin-arch.md:849]: I suggest to make some compromise for the above proposed text change.

**Review by boucadair — APPROVED** [review]

**Review by cfjanz — APPROVED** [review]

**Pull request merged**